Data/ByteString/Short/Internal.hs

```diff
@@ -1103,6 +1129,7 @@ breakEnd p = \sbs -> splitAt (findFromEndUntil p sbs) sbs
 -- @since 0.11.3.0
 break :: (Word8 -> Bool) -> ShortByteString -> (ShortByteString, ShortByteString)
 break = \p -> \sbs -> case findIndexOrLength p sbs of n -> (take n sbs, drop n sbs)
+{-# INLINE break #-}
```

Comment:
This probably shouldn't be inlined with zero arguments. (But it is probably almost never applied to zero arguments anyway.)

```suggestion
break p = \sbs -> case findIndexOrLength p sbs of n -> (take n sbs, drop n sbs)
{-# INLINE break #-}
```

Reply:
Does it make a difference? My understanding was that for inlining, the best strategy is to lambda all parameters. If no optimization fires, then doh.

Sharing won't help the consumer here anyway?

Reply:
Well, it is the best strategy with respect to inlining in as many situations as possible. But my point is that there does not seem to be any plausible benefit to inlining `break` unless it is applied to at least one argument, while there is a very clear cost to code size. I don't think it makes much difference, mainly because `break` without arguments is probably very uncommon.